### ISSUE TITLE
RegExpRouter returns not found

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#1140]
In this case, RegExpRouter returns a "not found" response, which is not expected.

```ts
app.get('/webhooks/github', async (c, next) => {
  await next()
})

app.get('/webhooks/:service', (c) => {
  return c.text(`service: ${c.req.param('service')}`)
})
```

```
GET /webhooks/github => 404 Not Found
```

If using TrieRouter, it returns a 200 response, though returning `undefined` is another issue.

```
GET /webhook/github => service: undefined
```

**Comment by usualoma** [comment]
Hi @yusukebe!

I'm sorry for not getting back to you sooner.

In that case, the expected result is for RegExpRouter to throw UnsupportedPathError.
RegExpRouter cannot handle this pattern because it needs to find one path in one regular expression match.

The PR to be corrected to return UnsupportedPathError will be #1146.

**Comment by yusukebe** [comment]
Thanks, I'll see it later!
